### PR TITLE
Update to the jslib.k6.io contribution guidelines

### PR DESCRIPTION
**Pull request by oleiade**
Here are the updated contribution guidelines as decided by the k6 open-source team. Kept it short and to the point on purpose. Please, let me know if you had something different in mind, or can think of improvements 🙇🏻

### CODE REVIEW

**Comment by oleiade** [comment]
Note that if no reviews or red flags are raised in the upcoming days, I'll just go ahead and merge this 🙇🏻

**Review by codebien — APPROVED** [review]

File: README.md
Comment (olegbespalov) [inline README.md:174]: :thinking: does that mean that any contributor will merge anything without the approval of any maintainer, and it will be deployed? :thinking:

Comment (oleiade) [reply README.md:174]: That's my understanding of what was decided by the k6 team: this repository mostly holds build files, we should also move whatever actual js code is in here to dedicated repositories, and we should give the flexibility to contributors to publish new versions of their jslib without having us blocking them.

Now, I also don't necessarily feel at ease with this change as is:  it could lead to somewhat of a security issue. Considering that anyone could push whatever package they wish without any other watch, they could also push malicious code (although I'm not certain what the scope and attack vectors could be in the context of k6+goja+jslib), but still.

Maybe a safer first approach is to indeed add "official" maintainers, and later add a bot such as the one in the definitely typed repo which checks that a new version's PR satisfies some predicate, pings maintainers for review, and allows the person to merge once.

Another alternative would be to allow only users from within the Grafana organization to push with reviews and to still require a review for externals.

What Do You Think?

Comment (olegbespalov) [reply README.md:174]: Yes, you got my concern right :hugs: 

>That's my understanding of what was decided by the k6 team:

For some reason I had the impression that we are extending the maintainers' list or allowing merging PRs without approvals only for grafana/k6 members.

So whatever the policy will be, I believe that something that has production deployment should have a gatekeeper.

Comment (oleiade) [reply README.md:174]: Thanks for the clarification on your assumptions folks 🙇🏻 I'm glad I double checked 🤝

Comment (mstoykov) [reply README.md:174]: from what I know nobody external can *merge* PRs so this seems like an unlikely thing to happen. 

Arguably though the section should be renamed as I don't think we have "contributors" - all the APIs here are done by us. And contributing will entail creating a different repo with the code and then asking us to host the final js files here. This seems unlikely to happen (and has not happen so far :thinking: ). It arguably is not even needed as the users can just host them on github (or wherever). Contributing something like that will also require documentation and so and so forth at which point I would argue we will push back on this unless we decide to adopt it - at which point it is no longer outside contribution ;).

tl;dr - can we explain that there are no "contributions" and that merging PR bumping versions is not required to have reviewers. Again only maintainers can merge PRs either way.

Comment (olegbespalov) [reply README.md:174]: >from what I know nobody external can merge PRs so this seems like an unlikely thing to happen.

If that's the case I'm absolutely fine with that. What confuses me is that it's a public repo and we introducing contributing section.

Comment (oleiade) [reply README.md:174]: @mstoykov those are fair points, I'm glad we got to (all together) clarify this.

>  What confuses me is that it's a public repo, and we introducing contributing section. So yeah, maybe we just need to stick with another name.

@olegbespalov I'd be interested in the reason leading to your confusion? I've actually double-checked, and most of the **public** [100 top repositories](https://github.com/EvanLi/Github-Ranking/blob/master/Top100/Top-100-stars.md) on GitHub do have contribution guidelines in their README (vue, react, TensorFlow, bootstrap, you name it).

If that's satisfying for everyone, moving forward with this PR I will:
* Remove the **contributing** section.
* Add a note somewhere in the description part of the README mentioning that PRs bumping versions of already supported jslibs do not need reviewers.
* Add a note somewhere in the description part of the README mentioning that PRs changing the jslib repository/webpages/project or adding a library would still need to be reviewed by a maintainer of the repo.

Now that doesn't really address the need I had which triggered this work initially: when submitting a PR that's not a version bump, who should be included as maintainers? As of today, this is unclear to me, and most PRs have been relying on @sniku or @ppcano, but on the other hand, with them being so busy, that (rightfully) doesn't really scale. Thus I would like to somehow clarify who should be the "maintainers" of the repo. In the sense of the people to turn to when needing a review.

I believe there are a bunch of options to clarify this down the road:
- [codeowner file](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners)
- "just" list one or two people to include as a reviewer as a default
- having a script automating picking k6 devs and frontend devs at random as reviewers, like in k6 (the frontend team on top of that).

Comment (olegbespalov) [reply README.md:174]: >I'd be interested in the reason leading to your confusion? I've actually double-checked, and most of the public [100 top repositories](https://github.com/EvanLi/Github-Ranking/blob/master/Top100/Top-100-stars.md) on GitHub do have contribution guidelines in their README (vue, react, TensorFlow, bootstrap, you name it).

All these projects that you mention have an established process of contribution to it. Just to be clear, by contributor here, I mean anybody outside Grafana. And I believe none of these projects allows us to merge anything without the explicit approval of the maintainer (like we advised in that PR). 

>If that's satisfying for everyone, moving forward with this PR I will:

all that you mentioned sounds good to me :+1: 

>Now that doesn't really address the need I had which triggered this work initially: when submitting a PR that's not a version bump, who should be included as maintainers? 

I had an impression during our core sync that you also volunteered for this, but maybe I'm wrong here. And maybe we can ask the frontend team if this will be something that they are also interested in reviewing as part of open-source initiatives.

File: .circleci/config.yml
Comment (codebien) [inline .circleci/config.yml:81]:
```suggestion
              only: main
```

**Review by codebien — CHANGES_REQUESTED** [review]

**Review by codebien — APPROVED** [review]
LGTM

**Review by olegbespalov — APPROVED** [review]
:+1:

**Review by mstoykov — APPROVED** [review]
LGTM!

**Comment by oleiade** [comment]
In an attempt to also address somewhat automatically assigning reviewers to PRs in this repo, I've added two things to the Pull Request:
* A PR auto-assign GitHub action setup which picks up automatically two reviewers. One from the k6 dev team, one from the k6 frontend engineers who contributed to the project before. If the PR is labeled with the `version bump` label, or if the PR contains the WIP, update, or updating or bump keywords, no reviewers will be picked.
* A PR template, which leads the submitter through a [DefinetlyTyped inspired](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/61666) checklist using the same steps as described in the README, but also trying to ensure the user does label their PR accordingly. 

I'm completely open to moving these changes it into a different PR, just let me know what your preference is 👍🏻

File: .github/pr-auto-assign-config.yml
Comment (codebien) [inline .github/pr-auto-assign-config.yml:24]: What is the default logic without any label?

Comment (oleiade) [reply .github/pr-auto-assign-config.yml:24]: If I'm to believe, the line of code that handles this case, I believe, the review auto-assignment is to be skipped. This means, that no labeling leads to no review requested. Not sure how to go about that 🤔 If I understand the logic leading into the if statement in the auto-assign code, I believe that by not adding the `include` section, the skip doesn't happen though 🤔

**Review by olegbespalov — APPROVED** [review]

**Pull request merged**
